src/main/java/org/openlmis/referencedata/service/export/DataImportService.java

```
@@ -0,0 +1,64 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.referencedata.service.export;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.openlmis.referencedata.dto.BaseDto;
+import org.openlmis.referencedata.util.FileHelper;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+public class DataImportService {
+
+  @Autowired
+  private FileHelper fileHelper;
+
+  @Autowired
+  private BeanFactory beanFactory;
+
+  /**
+   * Imports the data from a ZIP with CSV files.
+   *
+   * @param zipFile ZIP archive being imported.
+   */
+  @Transactional
+  public List<BaseDto> importData(MultipartFile zipFile) {
+    Map<String, InputStream> fileMap = fileHelper.convertMultipartFileToZipFileMap(zipFile);
+
+    List<BaseDto> result = new ArrayList<>();
+    for (Map.Entry<String, InputStream> entry: fileMap.entrySet()) {
+      if (!beanFactory.containsBean(entry.getKey())) {
+        throw new NoSuchBeanDefinitionException(
+            "Bean for parsing " + entry.getKey() + " not found!");
+      }
+
+      DataImportPersister<?, ?, ? extends BaseDto> persister =
+          beanFactory.getBean(entry.getKey(), DataImportPersister.class);
```

Just to make sure - `getBean` throws any exception if the bean is not found or returns null?

---

You're right, beanFactory.getBean() is throwing exception instead of returning null, I changed the code to handle this one correctly.